crates/iceberg/src/spec/partition.rs

```diff
@@ -69,6 +71,30 @@ impl PartitionSpec {
                 .iter()
                 .all(|f| matches!(f.transform, Transform::Void))
     }
+
+    /// Returns the partition type of this partition spec.
+    pub fn partition_type(&self, schema: &Schema) -> Result<StructType, Error> {
```

Comment:
Could we add some ut for this? What should be the return type if no partition field?

Reply:
> What should be the return type if no partition field?

For now, it return a empty struct type